src/screens/dialogs/upload-file-dialog.tsx

```diff
@@ -0,0 +1,96 @@
+import { useContext, useRef, useState } from "react";
+import toast from "react-hot-toast";
+import { uploadfileService } from "../../services/files/upload-file.service";
+import { Dialog } from "../../components/Dialog";
+import { Loader } from "lucide-react";
+import {
+  FilesDialogsContext,
+  AuthContext,
+  AVAILABLE_DIALOGS,
+  FilesContext
+} from "../../context/index";
+
+interface INewFile {
+  isFile: boolean;
+  isReady: boolean;
+  name: string;
+  size: number;
+  uuid: string;
+}
+
+export const UploadFileDialog = () => {
+  // Dialog state
+  const { closeDialog, dialogsVisibilityState } =
+    useContext(FilesDialogsContext);
+  const [isUploading, setIsUploading] = useState(false);
+  const isOpen = dialogsVisibilityState[AVAILABLE_DIALOGS.UPLOAD_FILE];
+  const { addFiles, currentDirectory } = useContext(FilesContext);
+  const { session } = useContext(AuthContext);
+
+  const files: INewFile[] = [];
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  const uploadFile = async () => {
+    setIsUploading(true);
+    const fileInput = fileInputRef.current;
```

Comment:
You didn't add the conditional. Even if the button shows a loader, the user is still able to click on it and start a new request which can affect our bandwidth.